test/integration/apiserver/admissionwebhook/reinvocation_test.go

```diff
@@ -194,19 +197,71 @@ func TestWebhookReinvocationPolicy(t *testing.T) {
 		}
 	}
 
+	_, err = client.CoreV1().Pods("default").Create(reinvocationMarkerFixture)
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	for i, tt := range testCases {
 		t.Run(tt.name, func(t *testing.T) {
-			recorder.Reset()
+			upCh := recorder.Reset()
 			ns := fmt.Sprintf("reinvoke-%d", i)
 			_, err = client.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns}})
 			if err != nil {
 				t.Fatal(err)
 			}
 
-			for i, webhook := range tt.webhooks {
-				defer registerWebhook(t, client, fmt.Sprintf("admission.integration.test%d", i), webhookServer.URL+webhook.path, webhook.policy, webhook.objectSelector)()
+			webhooks := []admissionv1beta1.MutatingWebhook{}
+			for j, webhook := range tt.webhooks {
+				name := fmt.Sprintf("admission.integration.test.%d.%s", j, strings.TrimPrefix(webhook.path, "/"))
+				fail := admissionv1beta1.Fail
+				endpoint := webhookServer.URL + webhook.path
+				webhooks = append(webhooks, admissionv1beta1.MutatingWebhook{
+					Name: name,
+					ClientConfig: admissionv1beta1.WebhookClientConfig{
+						URL:      &endpoint,
+						CABundle: localhostCert,
+					},
+					Rules: []admissionv1beta1.RuleWithOperations{{
+						Operations: []admissionv1beta1.OperationType{admissionv1beta1.OperationAll},
+						Rule:       admissionv1beta1.Rule{APIGroups: []string{""}, APIVersions: []string{"v1"}, Resources: []string{"pods"}},
+					}},
+					ObjectSelector:          webhook.objectSelector,
+					FailurePolicy:           &fail,
+					ReinvocationPolicy:      webhook.policy,
+					AdmissionReviewVersions: []string{"v1beta1"},
+				})
+			}
+
+			cfg, err := client.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Create(&admissionv1beta1.MutatingWebhookConfiguration{
+				ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("admission.integration.test-%d", i)},
+				Webhooks:   webhooks,
+			})
+			if err != nil {
+				t.Fatal(err)
 			}
 
+			// wait until new webhook is called the first time
+			if err := wait.PollImmediate(time.Millisecond*5, wait.ForeverTestTimeout, func() (bool, error) {
+				_, err = client.CoreV1().Pods("default").Patch(reinvocationMarkerFixture.Name, types.JSONPatchType, []byte("[]"))
+				select {
+				case <-upCh:
+					return true, nil
+				default:
+					t.Logf("Waiting for webhook to become effective, getting marker object: %v", err)
+					return false, nil
+				}
+			}); err != nil {
+				t.Fatal(err)
+			}
+
+			defer func() {
```

Comment:
should defer the delete right after creating... if the poll fails, we don't want to leave the webhook registered to pollute later tests